kongswap_adaptor/src/balances.rs

```diff
@@ -287,11 +300,18 @@ impl ValidatedBalances {
         // should not change more than the expected amount. Otherwise,
         // it means that by mistake more tokens than expected are
         // transferred to the external.
-        if balance_after.abs_diff(balance_before) > transferred_amount {
-            balance_book.suspense += balance_before.abs_diff(balance_after) - transferred_amount;
+        let manager_balance_delta = logged_saturating_sub(balance_before, balance_after);
+        if manager_balance_delta > transferred_amount {
+            balance_book.suspense = logged_saturating_add(
+                balance_book.suspense,
+                logged_saturating_sub(manager_balance_delta, transferred_amount),
+            );
         }
     }
 
+    // transferred_amount is the amount withdrawn from the external.
+    // Which means the amount received by the manager should be:
+    // transferred_amount - ledger fee
     pub(crate) fn find_withdraw_discrepency(
```

Comment:
```suggestion
    // This should equal to `balance_after - balance_before`.
    // Otherwise, there is a discrepancy, which this function records as the `suspense` balance.
    pub(crate) fn find_withdraw_discrepency(
```

Reply:
Btw, doc strings should start with `///`